src/components/backend-ai-signup.ts

```diff
@@ -199,47 +218,72 @@ export default class BackendAiSignup extends BackendAIPage {
     //this.shadowRoot.querySelector('#signup-button').setAttribute('disabled', 'true');
   }
 
+  _clearUserInput() {
+    this._toggleInputField(true);
+    const inputFields: Array<string> = ['#id_user_email', '#id_user_name', '#id_token', '#id_password1', '#id_password2'];
+    inputFields.map((el: string) => {
+      this.shadowRoot.querySelector(el).value = "";
+    });
+    this.shadowRoot.querySelector('#signup-button-message').textContent = 'Signup';
+  }
+
+  _toggleInputField(isActive: boolean) {
+    if (isActive) {
+      this.shadowRoot.querySelector('#id_user_name').removeAttribute('disabled');
+      this.shadowRoot.querySelector('#id_token').removeAttribute('disabled');
+      this.shadowRoot.querySelector('#signup-button').removeAttribute('disabled');
+    } else {
+      this.shadowRoot.querySelector('#id_user_name').setAttribute('disabled', 'true');
+      this.shadowRoot.querySelector('#id_token').setAttribute('disabled', 'true');
+      this.shadowRoot.querySelector('#signup-button').setAttribute('disabled', 'true');
+    }
+  }
+
   _signup() {
+    const inputFields: Array<string> = ['#id_user_email', '#id_user_name', '#id_token', '#id_password1', '#id_password2'];
+    let inputFieldsValidity: Array<boolean> = inputFields.map((el: string) => {
+      this.shadowRoot.querySelector(el).reportValidity();
+      return this.shadowRoot.querySelector(el).checkValidity();
+    })
+
     let approved = (this.shadowRoot.querySelector('#approve-terms-of-service') as HTMLInputElement).checked;
     if (approved === false) {
-      this.notification.text = "Please read and agree with the terms of service to proceed.";
-      this.notification.show();
-      return;
-    }
-    let password1 = (this.shadowRoot.querySelector('#id_password1') as HTMLInputElement).value;
-    let password2 = (this.shadowRoot.querySelector('#id_password2') as HTMLInputElement).value;
-    if (this.shadowRoot.querySelector("#id_password1").getAttribute("invalid") !== null) {
-      this.notification.text = "Password must contain at least one alphabet, one digit, and one special character";
+      this.notification.text = _text("signup.RequestAgreementTermsOfService");
       this.notification.show();
       return;
     }
-    if (password1 !== password2) {
-      this.notification.text = 'Password mismatch. Please check your password.';
-      this.notification.show();
+
+    // if any input is invalid, signup fails.
+    if (inputFieldsValidity.map((el: boolean) => {
+      if (!el) {
+        return !el;
+      }
+    })) {
```

Comment:
Isn't this `if` statement always `true`? The return of `map` is an `Array`, and an `Array` is always `true` even if it is empty. For example, the following code will log `foo`.

```javascript
if ([false]) console.log('foo');
```